x/sync/db.go

```diff
@@ -3,12 +3,23 @@
 
 package sync
 
-import "github.com/ava-labs/avalanchego/x/merkledb"
-
-type DB interface {
-	merkledb.Clearer
-	merkledb.MerkleRootGetter
-	merkledb.ProofGetter
-	merkledb.ChangeProofer
-	merkledb.RangeProofer
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/maybe"
+
+	pb "github.com/ava-labs/avalanchego/proto/pb/sync"
+)
+
+type DBSyncClient interface {
+	GetRootHash(ctx context.Context) (ids.ID, error)
+	HandleRangeProofResponse(ctx context.Context, request *pb.SyncGetRangeProofRequest, responseBytes []byte) (maybe.Maybe[[]byte], error)
+	HandleChangeProofResponse(
+		ctx context.Context,
+		request *pb.SyncGetChangeProofRequest,
+		responseBytes []byte,
+	) (maybe.Maybe[[]byte], error)
+	Error() error
+	Clear() error
 }
```

Comment:
Thoughts on this interface are:
1. The abstraction boundary isn't clear, we couple the database w/ how responses are handled
2. `Error` is interesting, we're expecting our sync package to just keep checking this return value. We could probably get rid of this from the interface if we moved this information somewhere else... we would only fatal during response handling so does it seem reasonable to see if it's possible to expose that information to that API?